### ISSUE TITLE
Running an Adonis App on an Ubuntu 14.04 Live Server?

### ISSUE THREAD

**Issue by WesThorburn** [adonisjs/adonis-framework#340]
This was the only piece of adonis live server set up documentation I could find: https://www.vultr.com/docs/adonis-js-setup-on-ubuntu-14

I have an Ubuntu 14.04 server, and I'm trying to get my adonis app running on it. According to **adonis check** my Node.JS version is 7.0.0, and my npm version is 3.10.8. I've pushed all my project files into the /var/www/[project-directory]. I installed all the dependencies and I'm able to run **nodemon --watch app server.js** from within the project directory.

When I try to access the app via the live URL, i'm simply shown the contents of the public directory because there is no index.html/index.js file. In the .env file on my server, the HOST value is set to 'localhost', the PORT is 8000 and the NODE_ENV is still set to 'development.'

I'm sure I'm missing a few steps here, but I just can't seem to find any documentation on this process. I should add that I have this app working flawlessly on my local machine. Any help would be appreciated.

**Comment by thetutlage** [comment]
Can u share the routes file of your application?

**Comment by WesThorburn** [comment]
I only have one route:

```
const Route = use('Route')

Route.on('/').render('game')
```

**Comment by thetutlage** [comment]
Yeah seems to be weird, are you accessing the port 8000 from the browser?

**Comment by WesThorburn** [comment]
No, I'm just accessing the URL. On my local machine, the app is accessible at 'localhost:8000' but on the live server, I shouldn't have to specify the port number in the URL right?

**Comment by thetutlage** [comment]
Nope, if you want to run the application by just specifying the domain name then it needs to run on PORT 80, otherwise you have to enter the port number.

The best practice is to run node.js app on an internal port like `8000` or `3333` etc and then use `nginx` to do a reverse proxy. You can learn more about Deploying node.js app using nginx reverse proxy on google

**Comment by WesThorburn** [comment]
Many of the guides I've found so far are specialized for multi-server scaling. Do you have any specific resources you can recommend? You've done an excellent job with the Adonis docs so far, it'd be great if there were some deployment docs too.

Also, I have to ask, what's the 'normal' way of deploying an Adonisjs app on a live server? I feel like I'm having a lot of trouble trying to accomplish what should be a very common task here. I'm from a PHP background, mainly with the Laravel framework so node is quite new to me.

**Comment by RomainLanz** [comment]
Hey @WesThorburn,

You should use [NGINX Reverse Proxy](https://www.nginx.com/resources/admin-guide/reverse-proxy/). It's very simple to configure it.

1. Install a process manager to run your Node.js application. ([pm2](http://pm2.keymetrics.io/) or whatever your want)
2. [Install Nginx](https://nginx.org/en/docs/install.html)
3. Configure the Reverse Proxy.
  ```
location / {
    proxy_pass http://127.0.0.1:333;
}
  ```

**Comment by thetutlage** [comment]
@WesThorburn This may help https://www.digitalocean.com/community/tutorials/how-to-set-up-a-node-js-application-for-production-on-ubuntu-14-04.

Also closing the issue, since not actionable and feel free to discuss it on [Gitter](https://gitter.im/adonisjs/adonis-framework)

**Comment by WesThorburn** [comment]
Alright, I solved it. For anyone else struggling to get their adonis app running on their Ubuntu server, here is how I did it.

- If you haven't already, install Nginx and PM2. 

- Push your project files to the _/var/www/project-dir_ directory.

- Install all the dependencies and ensure you're successfully able to run the adonis app on the server from the command line. Eg: **sudo pm2 start server.js**

- Take note of the ip address port number that your application is running on. For example: **127.0.0.1:8000**

- Cd into the /etc/nginx/sites-enabled directory and edit the default file using an editor. Eg: **sudo nano /etc/nginx/sites-enabled/default**

- The default file will have a lot of default configurations and comments, just ensure the following key configurations are present:

```
server {
     root /var/www/project-dir/public;
     location / {
          proxy_pass http://127.0.0.1:8000;
     }
}
```
- Ensure you update the the directory, ip address and port to match your application's configuration. 

- Save and exit out of the default file. Then apply the new changes by running: **sudo service nginx reload**

Overall, not a difficult process, but figuring out what needed to be done was a little tricky. I noticed in the Gitter there were a few other users asking for help on this exact issue. It'd be great if the adonis documentation covered this as well.
